providers/implementations/keymgmt/ml_kem_kmgmt.c

```diff
@@ -440,9 +444,14 @@ static void *ml_kem_gen(void *vgctx, OSSL_CALLBACK *osslcb, void *cbarg)
     int genok = 0;
 
     if (gctx == NULL
-        || (key = ml_kem_new(gctx->libctx, gctx->propq, gctx->variant)) == NULL)
+        || (key = ml_kem_new(gctx->libctx, gctx->propq, gctx->variant)) == NULL
+        || (gctx->selection & OSSL_KEYMGMT_SELECT_KEYPAIR) ==
+            OSSL_KEYMGMT_SELECT_PUBLIC_KEY)
```

Comment:
Please exchange these two conditions, otherwise `key` could be leaked.

Reply:
Done.